engine/crates/federated-graph/src/lib.rs

```diff
@@ -19,27 +19,27 @@ pub enum FederatedGraph {
     V1(FederatedGraphV1),
     V2(FederatedGraphV2),
     V3(FederatedGraphV3),
+    Sdl(String),
 }
 
 impl FederatedGraph {
-    pub fn into_sdl(self) -> Result<String, std::fmt::Error> {
-        render_sdl(self)
-    }
-
-    #[deprecated(note = "Use into_sdl() instead")]
-    pub fn to_sdl(&self) -> Result<String, std::fmt::Error> {
-        self.clone().into_sdl()
+    pub fn from_sdl(sdl: &str) -> Result<FederatedGraph, DomainError> {
+        Ok(FederatedGraph::Sdl(sdl.to_owned()))
     }
 
-    pub fn from_sdl(sdl: &str) -> Result<FederatedGraph, DomainError> {
-        from_sdl(sdl)
+    pub fn into_federated_sdl(self) -> String {
+        match self {
+            FederatedGraph::Sdl(sdl) => sdl,
+            other => render_federated_sdl(&other.into_latest()).unwrap(),
+        }
     }
 
-    pub fn into_latest(self) -> FederatedGraphV3 {
+    pub fn into_latest(self) -> FederatedGraphV4 {
```

Comment:
if `FederatedGraphV4` isn't meant to be stored, I would rename `FederatedGraph` to `VersionedFederatedGraph` and `FederatedGraphV4` to `FederatedGraph`

Reply:
Agree we might as well do it now. Pushing that in a minute.